File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organisation/event/OrganizationEditDefaultAddressEventDto.java
package de.helfenkannjeder.helfomat.api.organisation.event;

import de.helfenkannjeder.helfomat.api.organisation.AddressDto;
import de.helfenkannjeder.helfomat.core.organisation.OrganisationId;

/**
 * @author Valentin Zickner
 */
public class OrganizationEditDefaultAddressEventDto extends OrganizationEventDto {
    private AddressDto address;

    OrganizationEditDefaultAddressEventDto() {
    }

    public OrganizationEditDefaultAddressEventDto(OrganisationId organisationId, AddressDto address) {
        super(organisationId);
        this.address = address;
    }

    public AddressDto getAddress() {
        return address;
    }
}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditDefaultAddressEventDto.kt
package de.helfenkannjeder.helfomat.api.organization.event

import de.helfenkannjeder.helfomat.api.organization.AddressDto
import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
class OrganizationEditDefaultAddressEventDto(
    override val organizationId: OrganizationId,
    val address: AddressDto?,
    override val eventApplicable: Boolean
) : OrganizationEventDto {

    override fun <T> visit(visitor: OrganizationEventDtoVisitor<T>): T {
        return visitor.visit(this)
    }

}